src/PhpSpec/ObjectBehavior.php

```diff
@@ -31,9 +31,90 @@
  * @method void beConstructedWith(...$arguments)
  * @method void beConstructedThrough($factoryMethod, array $constructorArguments = array())
  * @method void beAnInstanceOf($class)
+ *
  * @method void shouldHaveType($type)
+ * @method void shouldNotHaveType($type)
+ * @method void shouldReturnAnInstanceOf($type)
```

Comment:
That is also true. It won't work for custom matchers and it won't work for matchers that depend on method names of the described class. What I wanted to achieve with this PR is to at least add those that are officially bundled anyways (and will most probably be the most used). But if you do not like this idea then feel free to close this PR :) I can totally understand your standpoint... But currently the only possible way to get some autocompletion with a PhpStorm plugin is to have those magic method annotation in the `Subject` class, because currently we cannot "add" those dynamically.